### ISSUE TITLE
[Feature request] Marco could use a desktop zoom feature

### ISSUE THREAD

**Issue by CHJ85** [mate-desktop/marco#549]
Hi there.
It'd be nice if Marco had a desktop zoom feature similar to the one in Compiz.
In my opinion, Compiz's desktop zoom is perfect, but the only one that actually does it right. Because it allows you to enable/disable mouse focus, mouse auto scale and so on.
And I really wish Marco had the same feature.

**Comment by vkareh** [comment]
This is a regularly asked-for feature, and I agree that we should have
_some_ solution, but it's not currently on the roadmap.

Some distros ship with
[`xzoom`](https://blends.debian.org/accessibility/tasks/x11), which you
can bind to a keyboard shortcut. On Ubuntu systems you can use
[magnus](https://github.com/stuartlangridge/magnus).

There might be others that I'm not aware of...

**Comment by CHJ85** [comment]
Thank you @vkareh.
I appreciate it.

**Comment by lukefromdc** [comment]
You can also get this right now in MATE by running compiz, which has a desktop
magnifier function.

**Comment by lukefromdc** [comment]
Sorry, didn't see the original post mentioning compiz before answering from email.
Before implementing something like this in marco we would need to ensure it could
be done on the CPU (compiz uses the GPU and openGL for the heavy lifting) without
being too heavy a load. We don't want marco with compositing on to become as heavy
as say, running compiz in LLVMpipe for CPU rendering, and that does put some limits on
what we can reasonably do.

In the old days, it seemed to me GNOME 2 devs deliberately kept metacity light while compiz
offered a very rich feature set for those machines capable of supporting it. It took years to get the
bugs out and to this day compiz is a beautiful but exceptionally complicated project.

**Comment by CHJ85** [comment]
@lukefromdc Right. Compiz is a complicated project because they essentially baked together 100s of features into 1. Not sure how heavy the Zoom plugin alone is on the GPU.
But what if you implement a similar zoom feature but turned off by default like XFCE?
Their zoom feature kinda annoys me because the mouse remains the same size.

**Comment by CHJ85** [comment]
It would however be nice if you could add this too your roadmap. The alternatives (xzoom and magnus) aren't very good alternatives.
The reason I don't wanna use Compiz is because their zoom plugin makes the web browser and file manager scroll while I zoom, which is a very annoying bug.

**Comment by josser900** [comment]
Now that Ubuntu Mate has dropped compiz for its LTS release it would be very nice if Marco could offer a full desktop zoom. I'm visually impaired and depend on zoom with color inversion. The other magnifiers (xzoom, magnus, etc) aren't suitable for this.
With previous Ubuntu Mate versions you could simply activate compiz (even before/during install) and start installing/using linux.

**Comment by raveit65** [comment]
Interesting. 
But compiz-0.9.x isn't dropped from repos and you can install it with you package-manager?

**Comment by josser900** [comment]
Yes, but it's a pain to install if you can't see what you're doing (several
packages +  configure ccsm to get it working properly).
Until now I could do super key, type term, enter, type compiz --replace,
enter (doesn't require sight ;) ). Super + scroll wheel to zoom, super+M to
invert colors and I could install and setup Ubuntu (so it would
automatically activate compiz on startup).
That's why I use Ubuntu Mate because it had this excellent accessibility.
And when we eventually switch over to Wayland then compiz won't work
anymore (don't know yet how much functionality Wayfire will inherit from
compiz).

**Comment by CHJ85** [comment]
As I see it, this is a must-have feature. Except for OpenBox, MATE is now the only DE that doesn't come with this feature.
I jumped to KDE for this very reason, which now uses less resources than MATE btw.
This is also something that the MATE team needs to keep in mind. While most of MATE's code is old and dated (hasn't been touched since Gnome 2), other DE's are becoming faster by keeping up with new code and removing old code.

**Comment by raveit65** [comment]
... unsubscribe from this topic...
